src/OrchardCore/OrchardCore.DisplayManagement/Extensions/ShapeTableProviderExtensions.cs

```diff
@@ -0,0 +1,19 @@
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.DisplayManagement.Descriptors;
+
+namespace OrchardCore.DisplayManagement;
+
+/// <summary>
+/// Provides an extension methods for <see cref="IShapeTableProvider"/>.
+/// </summary>
+public static class ShapeTableProviderExtensions
+{
+    /// <summary>
+    /// Registers required services for the specified <see cref="IShapeTableProvider"/>.
+    /// </summary>
+    /// <typeparam name="TShapeProvider">The <see cref="IShapeTableProvider"/> to be registered.</typeparam>
+    /// <param name="services">The <see cref="IServiceCollection"/>.</param>
+    public static IServiceCollection AddShapeProvider<TShapeProvider>(this IServiceCollection services)
```

Comment:
Probably AddShapeTableProvider so it's clear.

Reply:
Beleive it or not I'm not sure why it's called `ShapeTableProvider`?

Reply:
@sebastienros maybe able to provide more info on the naming. But, if you want to see an extension, it would have to be the same name as the interface